src/query/src/sql.rs

```diff
@@ -380,6 +487,71 @@ mod test {
         )
     }
 
+    #[test]
+    fn test_show_create_table() -> Result<()> {
+        let catalog_name = DEFAULT_CATALOG_NAME;
+        let schema_name = DEFAULT_SCHEMA_NAME;
+        let table_name = "test_table";
+        let schema = vec![
+            ColumnSchema::new(TABLE_COLUMN, ConcreteDataType::string_datatype(), false),
+            ColumnSchema::new(
+                CREATE_TABLE_COLUMN,
+                ConcreteDataType::string_datatype(),
+                false,
+            ),
+        ];
+        let data = vec![
+            Arc::new(StringVector::from(vec!["a".to_string()])) as _,
+            Arc::new(StringVector::from(vec!["table a".to_string()])) as _,
+        ];
+        let expected_columns = vec![
+            Arc::new(StringVector::from(vec!["test_table".to_string()])) as _,
+            Arc::new(StringVector::from(vec!["CREATE TABLE test_table (  table String  , create_table String engine=mock with(regions=1);".to_string()])) as _,
+        ];
+
+        show_create_table_test_by_schema(
+            catalog_name,
+            schema_name,
+            table_name,
+            schema,
+            data,
+            expected_columns,
+        )
+        // Ok(())
+    }
+
+    fn show_create_table_test_by_schema(
+        catalog_name: &str,
+        schema_name: &str,
+        table_name: &str,
+        schema: Vec<ColumnSchema>,
+        data: Vec<VectorRef>,
+        expected_columns: Vec<VectorRef>,
+    ) -> Result<()> {
+        let table_schema = SchemaRef::new(Schema::new(schema));
+        let catalog_manager =
+            prepare_describe_table(catalog_name, schema_name, table_name, table_schema, data);
```

Comment:
I prefer to rename this util function as it's now not only for testing describe table